File: 80-remove-duplicates-from-sorted-array-ii/remove-duplicates-from-sorted-array-ii.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        prev=2
        crr=2
        while(crr<len(nums)):
            if nums[crr]!=nums[prev-2]:
                nums[prev]=nums[crr]
                prev+=1
                crr+=1
            else:
                crr=crr+1
        return prev

File: 80-remove-duplicates-from-sorted-array-ii/remove-duplicates-from-sorted-array-ii.java
class Solution {
    public int removeDuplicates(int[] nums) {
        
        int prev=0,crr=0,last=Integer.MIN_VALUE,count=0;

        while(crr<nums.length){
            if(nums[crr]==last){
                count++;
            }
            else{
                last=nums[crr];
                count=1;
            }

            if(count<=2){
                nums[prev]=nums[crr];
                prev++;
            }
            crr++;
        }
        return prev;
    }
}